onnxruntime/contrib_ops/cuda/bert/rotary_embedding_impl.cu

```diff
@@ -105,10 +111,10 @@ Status LaunchRotaryEmbeddingKernel(
   ORT_ENFORCE(head_size <= max_threads_per_block,
               "Rotary embedding dim must be <= max_threads_per_block");
 
-  int tpb = (head_size + 31)/32*32;
+  int tpb = (head_size + 31) / 32 * 32;
 
   const dim3 block(tpb);
-  const dim3 grid(num_heads, sequence_length, batch_size);
+  const dim3 grid(num_heads, 1024, batch_size * (sequence_length / 1024 + 1));
```

Comment:
>1024 [](http://example.com/codeflow?start=29&length=4)

if seq length less than 1024, there are blocks that does nothing.

Reply:
if sequence length is 1024, half blocks are not needed since last two dims are 2 * batch_size * 1024, and only batch_size * 1024 is needed.